src/components/workstation/play-preview-button.tsx

```diff
@@ -0,0 +1,55 @@
+import { majorScale, minorScale } from "evergreen-ui";
+import { useCallback, useRef } from "react";
+import { useBoolean } from "utils/hooks/use-boolean";
+import { PlayButton, type PlayButtonProps } from "./play-button";
+
+interface PlayPreviewButtonProps
+    extends Omit<PlayButtonProps, "isPlaying" | "toggleIsPlaying"> {
+    fileUrl?: string;
+}
+
+const PlayPreviewButton: React.FC<PlayPreviewButtonProps> = (
+    props: PlayPreviewButtonProps
+) => {
+    const { disabled, fileUrl, isLoading, type, width, size } = props;
+    const audioRef = useRef<HTMLAudioElement>(null);
+    const {
+        value: isPlaying,
+        toggle: toggleIsPlaying,
+        setFalse: setIsPlayingFalse,
+    } = useBoolean();
+    const handlePlayClick = useCallback((isPlaying: boolean) => {
+        if (isPlaying) {
+            audioRef.current?.pause();
+            return;
+        }
+
+        audioRef.current?.play();
+    }, []);
+    return (
+        <>
+            <PlayButton
+                appearance="minimal"
+                disabled={disabled}
+                isLoading={isLoading}
+                isPlaying={isPlaying}
+                onClick={handlePlayClick}
+                size={size}
+                toggleIsPlaying={toggleIsPlaying}
+                type={type}
+                width={width}
+                {...props}
```

Comment:
One minor thing: since we're destructuring specific props we want to pass to `PlayButton`, can we use the `...remainingProps` syntax to spread just the props we haven't specifically pulled out? I think this will actually send `fileUrl` along and log a console warning:


```
register-console-error-toasts.ts:13 Warning: React does not recognize the `fileUrl` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `fileurl` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```

So something like:
```tsx
const { disabled, fileUrl, isLoading, type, width, size, ...iconButtonProps } = props;

// Other individually set props omitted for brevity
<PlayButton appearance="minimal" disabled={disabled} {...iconButtonProps}
```

Reply:
I updated the props being destructured, and simplified the props being passed to `PlayButton`. Good catch there.